### ISSUE TITLE
RunWorkflow Activity

### ISSUE THREAD

**Issue by sfmskywalker** [elsa-workflows/elsa-core#176]

**Comment by Andale-online** [comment]
Doesn't exist???
I need such in next development steps... Thought this is already done.

**Comment by Andale-online** [comment]
If not i'll do it today... got some time and RLY need it.

**Comment by sfmskywalker** [comment]
It's yours!

Some design notes:

* Name of the activity: `RunWorkflow`.
* The activity should take an optional **Input** workflow expression that evaluates to `Variables`.
* The activity should take an optional **correlationId** workflow expression that evaluates to `string`.
* If the workflow completes immediately (i.e. exits with the Finished status), the activity should return "Done" as the outcome.
* If the workflow invocation returns a workflow instance in the Executing status, the activity should return a HaltResult to halt the main workflow.
* There should be a message handler that listens for a yet to be created WorkflowFinished message. This handler should resume any halted workflows blocked by the RunWorkflow activity (correlated by workflow instance ID)


I already started on this: feel free to copy: https://github.com/elsa-workflows/elsa-core/blob/feature/run-workflow-activity/src/activities/Elsa.Activities.Workflows/Activities/RunWorkflow.cs

FYI - in the [feature/elsa-20](https://github.com/elsa-workflows/elsa-core/tree/feature/elsa-2.0) branch I started work on adding support for output & input for activities, allowing any activity to receive the output of the previous activity. I.e. do `return Done(myOutput)`.
Until that hits `develop`, just store the workflow output into the `Output` dictionary of the `RunWorkflow` activity.
